### ISSUE TITLE
[vLLM] Llama-3.2-3B produces all token_id 0 under TP on n300

### ISSUE THREAD

**Issue by kmabeeTT** [tenstorrent/tt-xla#3584]
## Problem

`test_structured_outputs_regex[n300]` fails with all token_id 0 output under TP. Investigation showed this is **not** a structured_decode ordering issue — the model itself produces garbage logits under TP.

Evidence: `test_greedy_determinism[n300]` (no grammar masking) also outputs all token_id 0 (`!!!!!!!!!!!!!!!!!!!!`). It "passes" only because the test checks determinism (same output 3x), not correctness.

## Reproduction

With `meta-llama/Llama-3.2-3B` and `enable_tensor_parallel: True` on n300:

```
# Structured output — FAILS (output doesn't match regex)
pytest -svv tests/integrations/vllm_plugin/sampling/test_sampling_params.py::test_structured_outputs_regex[n300]

# Greedy determinism — PASSES but output is garbage (all token_id 0)
pytest -svv tests/integrations/vllm_plugin/sampling/test_sampling_params.py::test_greedy_determinism[n300]
# Output: "!!!!!!!!!!!!!!!!!!!! " (all token_id 0, three runs identical)
```

The same model works correctly on `single_device` (no TP). The failure is specific to the TP path.

## Notes

- Llama-3.2-3B uses `tie_word_embeddings: True`, so `lm_head` is `VocabParallelEmbedding` (not `ParallelLMHead`). The `sharding_constraint_tensor` for logit replication at line 1263 is never reached.
- The `VocabParallelEmbedding` sharding in `vllm_distributed_utils.py:260` shards the weight as `(None, "x")` and registers a forward hook for the embedding output, but this hook only applies to the embedding lookup path, not when the same weight is used as the LM head.
- Related: #3579 (apply_grammar_bitmask on-device work)

**Comment by kmabeeTT** [comment]
Closing as duplicate of #3589. The root cause is not model-specific — it's torch.compile on sample_from_logits under SPMD mode producing all token_id 0. See #3589 for full investigation.
